source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -136,27 +137,37 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
                                                  Http::HeaderMap& headers, bool replacing_message,
                                                  const Checker& checker,
                                                  Counter& rejected_mutations,
+                                                 ProcessingEffect::Effect& effect,
                                                  bool remove_content_length) {
   // Check whether the remove_headers or set_headers size exceed the HTTP connection manager limit.
   // Reject the mutation and return error status if either one does.
   const auto result = responseHeaderSizeCheck(headers, mutation, rejected_mutations);
   if (!result.ok()) {
+    effect = ProcessingEffect::Effect::MutationRejectedSizeLimitExceeded;
     return result;
   }
 
+  // Set default value to None for effect
+  effect = ProcessingEffect::Effect::None;
+
   for (const auto& hdr : mutation.remove_headers()) {
     if (!Http::HeaderUtility::headerNameIsValid(hdr)) {
       ENVOY_LOG_PERIODIC(error, std::chrono::seconds(10),
                          "remove_headers contain invalid character, may not be removed.");
       rejected_mutations.inc();
+      effect = ProcessingEffect::Effect::InvalidMutationRejected;
       return absl::InvalidArgumentError(kInvalidCharacterInRemoveMsg);
     }
     const LowerCaseString remove_header(hdr);
     switch (checker.check(CheckOperation::REMOVE, remove_header, "")) {
-    case CheckResult::OK:
+    case CheckResult::OK: {
       ENVOY_LOG(trace, "Removing header {}", remove_header);
-      headers.remove(remove_header);
-      break;
+      // int removals;
+      int removals = headers.remove(remove_header);
+      if (removals > 0) {
+        effect = ProcessingEffect::Effect::MutationApplied;
```

Comment:
This is the first "effect" that I see that may be overridden by another one later (because there's no return right after it). Is the override intended?